nbclient/client.py

```diff
@@ -330,22 +380,47 @@ def start_kernel_manager(self):
         if self.km.ipykernel and self.ipython_hist_file:
             self.extra_arguments += ['--HistoryManager.hist_file={}'.format(self.ipython_hist_file)]
 
-        self.km.start_kernel(extra_arguments=self.extra_arguments, **kwargs)
+        # Support AsyncKernelManager
+        if inspect.iscoroutinefunction(self.km.start_kernel):
+            await self.km.start_kernel(extra_arguments=self.extra_arguments, **kwargs)
+        else:
+            self.km.start_kernel(extra_arguments=self.extra_arguments, **kwargs)
 
         self.kc = self.km.client()
         self.kc.start_channels()
         try:
             await self.kc.wait_for_ready(timeout=self.startup_timeout)
         except RuntimeError:
-            self.kc.stop_channels()
-            self.km.shutdown_kernel()
+            self._cleanup_kernel()
```

Comment:
Here we might use an async version of `_cleanup_kernel`, since we are in an async function.

Reply:
:+1: will do